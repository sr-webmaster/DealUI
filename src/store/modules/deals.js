import makeRestStore from '@freshinup/core-ui/src/store/utils/makeRestStore'

export const _getItemPath = ({ id }) => `/deals/v1/deals/${id}`

export default ({ items, item }, options = {}) => {
  let store = makeRestStore('deals', { items, item }, {
    itemsPath: () => `/deals/v1/deals`,
    itemPath: _getItemPath
  }, options)

  return {
    namespaced: true,
    ...store
  }
}
