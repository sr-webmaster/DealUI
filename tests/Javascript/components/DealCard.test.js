import { createLocalVue, mount, shallowMount } from '@vue/test-utils'
import { FIXTURE_DEALS } from '../../__data__/deals'
import Component from '~/components/DealCard.vue'

describe('Deals Card', () => {
  describe('Snapshots', () => {
    test('defaults', () => {
      const wrapper = mount(Component, {
        localVue: createLocalVue(),
        propsData: {
          deal: {
            id: 1,
            name: 'Test Name',
            customer: { first_name: 'First', last_name: 'Last', id: '123' },
            manager: { first_name: 'First', last_name: 'Last', id: '1234' },
            salesrep: { first_name: 'First', last_name: 'Last', id: '12345' },
            created_at: '45 Days',
            source: 'cars.com',
            type_id: 1,
            estimate: '25000',
            status_id: 1,
            status: { id: 1, name: 'Active' }
          }
        }
      })
      expect(wrapper.element).toMatchSnapshot()
      expect(wrapper.text()).toContain('Active')
    })
  })
  describe('clickDeal event', () => {
    test('occurs when clickDeal method is invoked', () => {
      const wrapper = shallowMount(Component, {
        localVue: createLocalVue(),
        propsData: {
          deal: {
            id: 1,
            name: 'Test Name',
            customer: { first_name: 'First', last_name: 'Last', id: '123' },
            manager: { first_name: 'First', last_name: 'Last', id: '1234' },
            salesrep: { first_name: 'First', last_name: 'Last', id: '12345' },
            created_at: '45 Days',
            source: 'cars.com',
            type_id: 1,
            estimate: '25000',
            status_id: 1,
            status: { id: 1, name: 'Active' },
            type: { id: 1, name: 'Mobile' }
          }
        }
      })
      wrapper.vm.clickDeal()

      // assert event has been emitted
      expect(wrapper.emitted().clickDeal).toBeTruthy()
      expect(wrapper.emitted().clickDeal).toHaveLength(1)

      // assert event payload
      expect(wrapper.emitted().clickDeal[0][0]).toEqual({
        id: 1,
        name: 'Test Name',
        customer: { first_name: 'First', last_name: 'Last', id: '123' },
        manager: { first_name: 'First', last_name: 'Last', id: '1234' },
        salesrep: { first_name: 'First', last_name: 'Last', id: '12345' },
        created_at: '45 Days',
        source: 'cars.com',
        type_id: 1,
        estimate: '25000',
        status_id: 1,
        status: { id: 1, name: 'Active' },
        type: { id: 1, name: 'Mobile' }
      })
    })
  })
  describe('Computed', () => {
    describe('titleColor', () => {
      test('active color', () => {
        const wrapper = shallowMount(Component, {
          localVue: createLocalVue(),
          propsData: {
            deal: FIXTURE_DEALS[0]
          }
        })
        expect(wrapper.vm.titleColor).toEqual('#5a51c2')
      })
      test('status id 2 color', () => {
        const wrapper = shallowMount(Component, {
          localVue: createLocalVue(),
          propsData: {
            deal: FIXTURE_DEALS[3]
          }
        })
        expect(wrapper.vm.titleColor).toEqual('#9e84b4')
      })
      test('default color', () => {
        const wrapper = shallowMount(Component, {
          localVue: createLocalVue(),
          propsData: {
            deal: {
              ...FIXTURE_DEALS[2],
              status: { id: 4, name: 'Status 4' }
            }
          }
        })
        expect(wrapper.vm.titleColor).toEqual('#e01a00')
      })
    })
  })
})
