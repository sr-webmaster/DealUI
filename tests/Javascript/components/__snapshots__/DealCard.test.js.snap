// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deals Card Snapshots defaults 1`] = `
<div
  class="v-card v-sheet theme--light"
  style="height: 100%;"
>
  <div
    class="v-card__title pa-0 ma-0"
  >
    <nav
      class="v-toolbar theme--dark"
      style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px); background-color: rgb(90, 81, 194); border-color: #5a51c2;"
    >
      <div
        class="v-toolbar__content"
        style="height: 64px;"
      >
        <div
          class="v-toolbar__title"
        >
          Active
        </div>
      </div>
    </nav>
  </div>
   
  <div
    class="v-card__text pt-0"
  >
    <h2
      class="my-3 font-weight-bold primary--text"
    >
      
      Test Name
    
    </h2>
     
    <hr
      class="v-divider theme--light"
    />
     
    <div
      class="layout row"
    >
      <div
        class="flex font-weight-bold text-uppercase caption pa-1 xs5"
      >
        
        Customer
      
      </div>
       
      <div
        class="flex font-weight-bold f-dealCard__highlight pa-1 xs7 text-xs-right"
      >
        
        First
        Last
        
        <br />
        
        #123
      
      </div>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
     
    <div
      class="layout row"
    >
      <div
        class="flex font-weight-bold text-uppercase caption pa-1 xs5"
      >
        
        Estimate
      
      </div>
       
      <div
        class="flex font-weight-bold f-dealCard__highlight pa-1 xs7 text-xs-right"
      >
        
        25000
      
      </div>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
     
    <div
      class="layout row"
    >
      <div
        class="flex font-weight-bold text-uppercase caption pa-1 xs5"
      >
        
        Age
      
      </div>
       
      <div
        class="flex pa-1 xs7 text-xs-right"
      >
        
        in 26 years
      
      </div>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
     
    <div
      class="layout row"
    >
      <div
        class="flex font-weight-bold text-uppercase caption pa-1 xs5"
      >
        
        Sales Rep
      
      </div>
       
      <div
        class="flex pa-1 xs7 text-xs-right"
      >
        
        First
        Last
      
      </div>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
     
    <div
      class="layout row"
    >
      <div
        class="flex font-weight-bold text-uppercase caption pa-1 xs5"
      >
        
        Source
      
      </div>
       
      <div
        class="flex pa-1 xs7 text-xs-right"
      >
        
        cars.com
      
      </div>
    </div>
  </div>
</div>
`;
